### ISSUE TITLE
test_batched_underfill_does_not_leak_the_topk_sentinel is not tie-safe: fails on CPU bfloat16 with torch 2.14

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4219]
### Summary

`TestScaleSpaceDetector::test_batched_underfill_does_not_leak_the_topk_sentinel` fails on CPU
`bfloat16` with torch 2.14.0. It is a **test** defect, not a detector defect: the test asserts that
the batched and the single-image top-K lists agree **positionally**, and that is not well defined
when responses tie — which in `bfloat16` they almost always do.

### Diagnosis

For `torch.manual_seed(0)`, `inp = torch.rand(2, 1, 64, 64)`, `ScaleSpaceDetector()` on CPU
`bfloat16`:

- responses are **bitwise identical** between the batched and unbatched runs (`max|diff| == 0`);
- exactly **two LAF rows differ, and they are swapped with each other**:

  ```
  row 3: response 0.00225830078125 (occurs 2x) -- centre batched [16.0, 14.0] vs unbatched [47.0, 46.0]
  row 4: response 0.00225830078125 (occurs 2x) -- centre batched [47.0, 46.0] vs unbatched [16.0, 14.0]
  ```

- the 500 responses take **16 distinct `bfloat16` values**, so ties are the norm, not the exception.

So both runs found the same set of keypoints with the same scores; `topk` merely broke a tie in a
different order for `B = 2` than for `B = 1`. `float16` and `float32` pass on the same input and
seed (`max|diff| == 0` on both), because they resolve the two responses apart.

### Scope

- Reproduces 3/3 on torch **2.14.0**, macOS/arm64, Python 3.11, CPU `bfloat16`.
- Not visible in PR CI: the `half-precision (bfloat16)` leg runs torch **2.9.1**, where the tie
  happens to break the same way for both batch sizes, and the leg is green.
- Not in `testing/known_failure_xfails/cpu-bfloat16.txt`.
- Pre-existing on `main`. Found while reviewing #4217; verified unrelated to it by checking out
  `kornia/feature/laf.py` at base `8db1499a` with everything else at that branch, where it fails
  identically 3/3.

### Suggested fix

The last two lines of the test are the problem:

```python
lafs1, resps1 = det(inp[:1])
self.assert_close(resps1[0], resps[0])
self.assert_close(lafs1[0], lafs[0])
```

The batched/unbatched agreement the test wants is a **set** property, not an ordering. Options, in
order of preference:

1. Compare order-insensitively — sort both by `(response, centre_x, centre_y)`, or match rows by
   centre before comparing.
2. Keep the positional comparison but choose an input whose responses do not tie in `bfloat16`, and
   say in a comment that this is load-bearing.
3. If neither, drop the batched-vs-unbatched half of the test into its own case so the sentinel
   assertions — which are the ones the test is named for and which pass — are not taken down with it.

Note the existing comment above those lines already anticipates a weaker version of this ("the
response convolutions pick batch-size-dependent kernels on some CPUs ... last-ULP-different
scores"); the actual failure mode is stronger than a tolerance issue, since the scores here are
identical and the *ordering* is what moves.

### Repro

```python
import torch
from kornia.feature import ScaleSpaceDetector

torch.manual_seed(0)
dt = torch.bfloat16
inp = torch.rand(2, 1, 64, 64, dtype=dt)
det = ScaleSpaceDetector().to(torch.device("cpu"), dt)
lafs, resps = det(inp)
lafs1, resps1 = det(inp[:1])
print("resp max|diff|:", (resps1[0].float() - resps[0].float()).abs().max().item())
print("laf  max|diff|:", (lafs1[0].float() - lafs[0].float()).abs().max().item())
```

```
resp max|diff|: 0.0
laf  max|diff|: 32.0
```

Posted on behalf of @ducha-aiki by Claude (Opus 5, 1M context).
